src/Testcontainers.Gitlab/GitlabBuilder.cs

```diff
@@ -0,0 +1,78 @@
+ namespace Testcontainers.Gitlab;
+ 
+ /// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+ [PublicAPI]
+ public sealed class GitlabBuilder : ContainerBuilder<GitlabBuilder, GitlabContainer, GitlabConfiguration>
+ {
+     /// <summary>
+     /// This is the default image for gitlab community edition.
+     /// </summary>
+     public const string GitlabImage = "gitlab/gitlab-ce";
+     /// <summary>
+     /// This port is used for http communication to gitlab instance.
+     /// </summary>
+
+     public const ushort GitlabHttpPort = 80;
+     /// <summary>
+     /// This port is used for ssh communication to gitlab instance.
+     /// </summary>
+     public const ushort GitlabSshPort = 22;
+
+     /// <summary>
+     /// Initializes a new instance of the <see cref="GitlabBuilder" /> class.
+     /// </summary>
+     public GitlabBuilder()
+         : this(new GitlabConfiguration())
+     {
+         DockerResourceConfiguration = Init().DockerResourceConfiguration;
+     }
+
+     /// <summary>
+     /// Initializes a new instance of the <see cref="GitlabBuilder" /> class.
+     /// </summary>
+     /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+     private GitlabBuilder(GitlabConfiguration resourceConfiguration)
+         : base(resourceConfiguration)
+     {
+         DockerResourceConfiguration = resourceConfiguration;
+     }
+
+     /// <inheritdoc />
+     protected override GitlabConfiguration DockerResourceConfiguration { get; }
+
+     /// <inheritdoc />
+     public override GitlabContainer Build()
+     {
+         Validate();
+         return new GitlabContainer(DockerResourceConfiguration);
+     }
+
+     /// <inheritdoc />
+     protected override GitlabBuilder Init()
+     {
+         return base.Init()
+             .WithImage(GitlabImage)
+             .WithPortBinding(GitlabHttpPort, true)
+             .WithPortBinding(GitlabSshPort, true)
+             .WithWaitStrategy(Wait.ForUnixContainer().UntilFileExists("/etc/gitlab/initial_root_password"))
+             .WithWaitStrategy(Wait.ForUnixContainer().UntilPortIsAvailable(80))
+             .WithWaitStrategy(Wait.ForUnixContainer().UntilPortIsAvailable(22))
+             .WithWaitStrategy(Wait.ForUnixContainer().UntilContainerIsHealthy())
+             .WithWaitStrategy(Wait.ForUnixContainer().UntilHttpRequestIsSucceeded(request => request.ForPath("/users/sign_in").ForStatusCode(HttpStatusCode.OK)));
```

Comment:
This overrides the previous wait strategies. You need to add the wait strategy once (you can chain the until members though).

Reply:
I've fixed the WaitStrategy so that the previous strategy will not be overwritten.